packages/protocol-http/src/extensions/http.ts

```diff
@@ -0,0 +1,42 @@
+import { HttpHandler } from "../httpHandler";
+import { HttpHandlerExtensionConfiguration } from "./httpExtensionConfiguration";
+
+/**
+ * @internal
+ */
+export const getHttpHandlerConfiguration = <HandlerConfig extends object = Record<string, unknown>>(
+  runtimeConfig: Partial<{ httpHandler: HttpHandler<HandlerConfig> }>
+) => {
+  let httpHandler = runtimeConfig.httpHandler!;
+  const httpHandlerConfigs: HandlerConfig = httpHandler.httpHandlerConfigs();
+  return {
+    setHttpHandler(handler: HttpHandler<HandlerConfig>): void {
+      httpHandler = handler;
+    },
+    httpHandler(): HttpHandler<HandlerConfig> {
+      return httpHandler;
```

Comment:
when setHttpHandler is called, `httpHandler()` will start to return its new value. However, the config's httphandler will not change. Is that intended?